[Ma Hanyang] iP

m-xrtin opened the pull request.

m-xrtin committed:
Renamed instances of "Duke" to "Tarnia"

Changed bot's name to "Tarnia"

m-xrtin committed:
Renamed instances of "Duke" to "Tarnia"

Changed bot's name to "Tarnia"

m-xrtin committed:
Merge branch 'master' of https://github.com/m-xrtin/ip

m-xrtin committed:
Provide Skeleton Greeting and Goodbye messages for bot

m-xrtin committed:
Updated Logo to fit our bot's name

m-xrtin committed:
Implemented the Echo Feature

Imported Scanner so that we can grab user input, and repeat after the user. Exits the program when the user inputs "bye", prints goodbye message and finishes program.

m-xrtin committed:
Updated Tarnia to be able to add tasks and output the existing list.

Also changed some formatting with \n as println already prints a new line, set list as 100 entries as prompt suggests input will not exceed 100.

m-xrtin committed:
Created Task Class to better manage list

Also fixed up the logic, changing from if/else statements to Switch/Case statements for simplicity

m-xrtin committed:
Updated to Java Coding Standards

JordanTwz commented on src/main/java/Tarnia.java:
```diff
@@ -0,0 +1,64 @@
+
+        Scanner in = new Scanner(System.in);
+        Task[] tasks = new Task[100];
+        int counter = 0;
```
Variable names should reveal intent. 
```suggestion
        int taskCounter = 0;
```

JordanTwz commented on src/main/java/Tarnia.java:
```diff
@@ -0,0 +1,64 @@
+                break;
+
+            case "mark":
+                int mark_index = Integer.parseInt(parts[1]) - 1;
```
mark_index, unmark_index use snake_case; Java conventions prefer markIndex, unmarkIndex.

```suggestion
                int markIndex = Integer.parseInt(parts[1]) - 1;
```

bennyy117 replied:
Any reason why `mark_index `and `unmark_index` are in snake_case? It prefers camelCase (`markIndex`, `unmarkIndex`). It might help with consistency across the project.

JordanTwz commented on src/main/java/Tarnia.java:
```diff
@@ -0,0 +1,64 @@
+        System.out.println("What can I do for you?");
+
+        Scanner in = new Scanner(System.in);
+        Task[] tasks = new Task[100];
```
Avoid magic numbers. Using a predefined constant helps to improve intent and self-documentation.

```suggestion
        Task[] tasks = new Task[MAX_TASKS];
```

bennyy117 replied:
Perhaps a constant `MAX_TASKS` could be used instead of the number `100`. This could make it clearer what this number represents and allow easier changes in the future.

JordanTwz commented on src/main/java/Tarnia.java:
```diff
@@ -0,0 +1,64 @@
+        int counter = 0;
+
+        while (true) {
+            String line = in.nextLine().trim();
```
By trimming off leading and trailing spaces, you prevent edge cases like "list " or " mark 2" from breaking your parser or being misinterpreted. Well done!

JordanTwz reviewed (commented):
Overall, this is a clear and functional implementation with thoughtful touches like .trim(), and with a few naming and readability refinements it will be even more maintainable and professional.

xelisce commented on src/main/java/Tarnia.java:
```diff
@@ -0,0 +1,64 @@
+            String[] parts = line.split(" ", 2);
+            String command = parts[0];
+
+            // Switch is more efficient for these tasks, changed from if/else statements
```
Avoid writing what you've changed in the comments. Instead, the reader should gain more information on what the code is doing. Referencing ["Write to the reader"](https://nus-cs2113-ay2526s1.github.io/website/se-book-adapted/chapters/codeQuality.html#write-to-the-reader).

tam308 commented on src/main/java/Tarnia.java:
```diff
@@ -0,0 +1,64 @@
+            String command = parts[0];
+
+            // Switch is more efficient for these tasks, changed from if/else statements
+            switch (command) {
```
perhaps the switch cases can be abstracted into methods?

tam308 commented on src/main/java/Tarnia.java:
```diff
@@ -0,0 +1,64 @@
+
+            case "unmark":
+                // Offsetting the index
+                int unmark_index = Integer.parseInt(parts[1]) - 1;
```
unmark_index here should be written in lowerCamelCase

xelisce commented on src/main/java/Tarnia.java:
```diff
@@ -0,0 +1,64 @@
+            case "list":
+                System.out.println("Here are the tasks in your list! :");
+                for (int i = 0; i < counter; i++) {
+                    System.out.println((i + 1) + ". "
```
This entire method `main()` is too long and it is difficult for the reader to quickly understand the flow of events. Instead, you can try abstracting some of the more "tedious" tasks to a different method. For example, the entire list case here could be its own method, `listTasks()`

xelisce commented on src/main/java/Tarnia.java:
```diff
@@ -0,0 +1,64 @@
+
+            case "mark":
+                int mark_index = Integer.parseInt(parts[1]) - 1;
+                tasks[mark_index].markDone();
```
`tasks[mark_index]` is repeated thrice. Consider making its own variable `Task unmarkTask = tasks[markIndex];`

xelisce commented on src/main/java/Tarnia.java:
```diff
@@ -0,0 +1,64 @@
+
+public class Tarnia {
+    public static void main(String[] args) {
+        System.out.println("Hello! I'm 𝓣𝓪𝓻𝓷𝓲𝓪 💔");
```
All the printing statements are cluttering up this method. The welcome and bye messages can also be put in its own methods.

tam308 replied:
perhaps these 2 strings can be in one line separated by a \n

tam308 replied:
personalisation is nice, but need to make sure the emoji and text can be rendered correctly on different environments

xelisce commented on src/main/java/Tarnia.java:
```diff
@@ -0,0 +1,64 @@
+                break;
+
+            case "bye":
+                System.out.println("BYE BYE. Hope to see you again soon! xo xo");
```
I like the personality your chatbot has!

xelisce commented on src/main/java/Tarnia.java:
```diff
@@ -0,0 +1,64 @@
+                tasks[counter] = new Task(line);
+                System.out.println("added for ya: "
+                        + "[" + tasks[counter].getStatusIcon() + "] "
+                        + tasks[counter].getDescription());
```
Good job in splitting up a long print line into separate lines!

tam308 commented on src/main/java/Tarnia.java:
```diff
@@ -0,0 +1,64 @@
+            case "mark":
+                int mark_index = Integer.parseInt(parts[1]) - 1;
+                tasks[mark_index].markDone();
+                System.out.println("Yayyyy :), I've marked this task as done:");
```
Nice personalisation of the bot!

bennyy117 commented on src/main/java/Tarnia.java:
```diff
@@ -0,0 +1,64 @@
+import java.util.Scanner;
+
+public class Tarnia {
+    public static void main(String[] args) {
```
Have you considered splitting the `main()` into smaller helper methods like `displayWelcome()`, `listTasks()`, `addTask()`, `markTask()`? It might make it easier for future readers to follow the flow.

bennyy117 commented on src/main/java/Tarnia.java:
```diff
@@ -0,0 +1,64 @@
+            case "unmark":
+                // Offsetting the index
+                int unmark_index = Integer.parseInt(parts[1]) - 1;
+                tasks[unmark_index].markUndone();
```
I noticed `tasks[unmark_index]` are accessed multiple times. Have you considered assigning them to a temporary variable like `Task markTask = tasks[markIndex]`;? It could reduce repetition and make the code easier to read.

bennyy117 reviewed (commented):
Overall, I like how you handled all the tasks in `main()`. Good job.

m-xrtin committed:
Updated Structure to be more SRP

m-xrtin committed:
Updated program to take in different types of tasks

Now can treat deadline, events, and todo separately

m-xrtin committed:
Created various error capture features

Now able to tell the user what they did wrong, either missing message or incorrect format for deadline and events (as well as how to fix)

m-xrtin committed:
Merge branch 'branch-Level-5'

m-xrtin committed:
Updated everything into one package

m-xrtin committed:
Merge branch 'branch-A-Packages'

syCHEN1645 commented on src/main/java/tarnia/CommandManager.java:
```diff
@@ -0,0 +1,40 @@
+        this.manager = manager;
+    }
+
+    public boolean handleCommand(String line) {
```
Good abstraction, looks very cneat and clear!

syCHEN1645 commented on src/main/java/tarnia/Messages.java:
```diff
@@ -0,0 +1,59 @@
+public class Messages {
+
+    public static void printHelloMessage() {
+        System.out.println("Hello! I'm 𝓣𝓪𝓻𝓷𝓲𝓪 💔");
```
Interesting! Just be careful with special characters, as they may not always show properly on other machines/terminals.

syCHEN1645 commented on src/main/java/tarnia/Ui.java:
```diff
@@ -0,0 +1,59 @@
+    }
+
+    public static void printOutOfRange(String command) {
+        System.out.println("Oopsie Daisies :/ The task index given for " + command + " is out of range.");
```
Very clear error message, nice job!

syCHEN1645 reviewed (commented):
Great job, keep it up!

m-xrtin committed:
Adding new methods to deal with deleting a task from the list.

m-xrtin committed:
Now saves and loads a textfile for all the tasks.

Takes care of missing folder by creating a new file.

m-xrtin committed:
Merge branch 'branch-Level-6'

m-xrtin committed:
Unable to rebranch properly, branched Level 7 on top of Level 6 instead of from Master :(

m-xrtin committed:
Merge remote-tracking branch 'origin/branch-Level-7'

m-xrtin committed:
Jar file now runnable

Also outputs a data folder inside of the ip_jar folder

m-xrtin committed:
Refactored the Messages class to be the UI, And TaskManager to be TaskList to better fit Assignment needs

m-xrtin committed:
Implemented the find command, which allows user to look for specific tasks (case insensitive) in the task list.

m-xrtin committed:
Added JavaDoc comments to at least half of the methods, and classes.

m-xrtin committed:
Update README.md

m-xrtin committed:
Update README.md

m-xrtin committed:
Update README.md

m-xrtin committed:
Update README.md

m-xrtin committed:
Update README.md

m-xrtin committed:
Update README.md

m-xrtin committed:
Update README.md

m-xrtin committed:
Update README.md

m-xrtin committed:
Final jar completed

m-xrtin committed:
Final Readme completed

m-xrtin committed:
Final product completed as a jar

m-xrtin committed:
Renamed to other file

m-xrtin committed:
Updated Readme

m-xrtin committed:
Update README.md
